components/client/src/main/java/com/hotels/styx/client/netty/connectionpool/NettyConnectionFactory.java

```diff
@@ -93,6 +103,15 @@ private NettyConnectionFactory(Builder builder) {
         });
     }
 
+    @Override
+    public CompletableFuture<Void> close() {
+        if (eventLoopGroup == GLOBAL_CLIENT_EVENT_LOOP) {
+            return completedFuture(null);
+        }
+
+        return fromNettyFuture(eventLoopGroup.shutdownGracefully());
```

Comment:
I don't know if I understand this code correctly: if the eventLoopGroup has been injected externally (and thus could be potentially shared among several different connection factories) we close it here.  That doesn't seem correct. 
Also we are initializing a  Default Global Event Loop that's not going to be used in most cases.

Reply:
Your criticism is justified. Let me see if event loop can be closed outside.

Reply:
Reading your comment again, and looking at code, I notice that `eventLoopGroup` is not really injected. It is created with the factory that has been externally injected.

But with memoization it may have the effect you describe: The same `eventLoopGroup` could end up being shared with different connection factories, and one of the factories may end up closing it on behalf of others.

This feels an incorrect usage of `NettyConnectionFactory` rather than an error in the `NettyConnectionFactory` itself.

Reply:
Oh, sure, that's right. In this case we might be safe ourselves as we don't try to close the connectionFactory except for healthchecks  - and the StyxHttpClient used in healthchecks is using a different eventLoopGroup than the forwarding path...  But it seems it could lead to some issues.

Reply:
Please check this commit: a5d70cd123635ce59fde8acea9f28bc460fe92b8.

It is a different approach and removes the `close` method. 

Now the Netty `eventLoopGroup` and accompanying channel type are injected from outside. They are never closed. The client event loop group is created when styx starts and stays alive until shut.

The Styx health checks & Styx standalone client are now permanently fixed on "global" event loop group without option to inject its own. This can be improved if necessary.